### PR TITLE
Migrate Match, error, default values in internal package

### PR DESCRIPTION
**Pull request by VijithaEkanayake**
## Purpose
$subject.

### CODE REVIEW

File: stdlib/internal/src/main/ballerina/internal/jwt_issuer.bal [hunk 1]
```diff
@@ -62,8 +66,9 @@ function createHeader(JwtHeader header) returns (string|error) {
 function createPayload(JwtPayload payload) returns (string|error) {
     json payloadJson = {};
     if (!validateMandatoryFields(payload)) {
-        error err = error("Mandatory fields(Issuer, Subject, Expiration time or Audience) are empty.");
-        return err;
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Mandatory fields(Issuer, Subject,
```
Comment (wggihan) [inline stdlib/internal/src/main/ballerina/internal/jwt_issuer.bal]: Shall we break this line and make the message part as a single line?

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 2]
```diff
@@ -42,51 +42,63 @@ public type JWTValidatorConfig record {
...
-    match getJWTComponents(jwtToken) {
-        string[] encodedJWT => encodedJWTComponents = encodedJWT;
-        error e => return e;
+    var jwtComponents = getJWTComponents(jwtToken);
+    if (jwtComponents is string[]) {
+        encodedJWTComponents = jwtComponents;
+    } else if (jwtComponents is error) {
+        return jwtComponents;
+    } else {
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Invalid JWT token"});
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]:
```suggestion
        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Invalid JWT token" });
```

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 3]
```diff
@@ -42,51 +42,63 @@ public type JWTValidatorConfig record {
...
-            (header, payload) = result;
-        }
+    string[] aud = [];
+    JwtHeader header = {alg: ""};
+    JwtPayload payload = {iss: "", sub: "", aud: aud, exp: 0};
+    var decodedJwt = parseJWT(encodedJWTComponents);
+    if (decodedJwt is (JwtHeader, JwtPayload)) {
+        (header, payload) = decodedJwt;
+    } else if (decodedJwt is error) {
+        return decodedJwt;
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]: Should we create a new error with INTERNAL_ERROR_CODE here?

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 4]
```diff
@@ -42,51 +42,63 @@ public type JWTValidatorConfig record {
...
     }
 }
 
 function getJWTComponents(string jwtToken) returns (string[])|error {
     string[] jwtComponents = jwtToken.split("\\.");
     if (jwtComponents.length() != 3) {
         log:printDebug("Invalid JWT token :" + jwtToken);
-        error err = error("Invalid JWT token");
-        return err;
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Invalid JWT token"});
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]:
```suggestion
        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Invalid JWT token" });
```

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 5]
```diff
@@ -176,35 +193,48 @@ function parsePayload(json jwtPayloadJson) returns (JwtPayload) {
 
 function validateJWT(string[] encodedJWTComponents, JwtHeader jwtHeader, JwtPayload jwtPayload, JWTValidatorConfig
 config) returns (boolean|error) {
+    if (!validateMandatoryJwtHeaderFields(jwtHeader)) {
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Mandatory field signing algorithm(alg)
+         is empty in the given JSON Web Token."});
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]:
```suggestion
         is empty in the given JSON Web Token." });
```

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 6]
```diff
@@ -176,35 +193,48 @@ function parsePayload(json jwtPayloadJson) returns (JwtPayload) {
...
+        return jwtError;
+    }
     if (!validateMandatoryFields(jwtPayload)) {
-        error err = error("Mandatory fields(Issuer, Subject, Expiration time or Audience) are empty in the given JSON Web Token.");
-        return err;
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Mandatory fields(Issuer,
+         Subject, Expiration time or Audience) are empty in the given JSON Web Token."});
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]:
```suggestion
         Subject, Expiration time or Audience) are empty in the given JSON Web Token." });
```

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 7]
```diff
@@ -176,35 +193,48 @@ function parsePayload(json jwtPayloadJson) returns (JwtPayload) {
...
+        return jwtError;
     }
     if (!validateSignature(encodedJWTComponents, jwtHeader, config)) {
-        error err = error("Invalid signature");
-        return err;
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Invalid signature"});
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]:
```suggestion
        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Invalid signature" });
```

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 8]
```diff
@@ -176,35 +193,48 @@ function parsePayload(json jwtPayloadJson) returns (JwtPayload) {
...
+        return jwtError;
     }
     if (!validateIssuer(jwtPayload, config)) {
-        error err = error("JWT contained invalid issuer name : " + jwtPayload.iss);
-        return err;
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "JWT contained invalid issuer name : " + jwtPayload.iss});
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]:
```suggestion
        error jwtError = error(INTERNAL_ERROR_CODE, { message : "JWT contained invalid issuer name : " + jwtPayload.iss });
```

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 9]
```diff
@@ -176,35 +193,48 @@ function parsePayload(json jwtPayloadJson) returns (JwtPayload) {
...
+        return jwtError;
     }
     if (!validateAudience(jwtPayload, config)) {
         //TODO need to set expected audience or available audience list
-        error err = error("Invalid audience");
-        return err;
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Invalid audience"});
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]:
```suggestion
        error jwtError = error(INTERNAL_ERROR_CODE, { message : "Invalid audience" });
```

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 10]
```diff
@@ -176,35 +193,48 @@ function parsePayload(json jwtPayloadJson) returns (JwtPayload) {
...
+        return jwtError;
     }
     if (!validateExpirationTime(jwtPayload, config)) {
-        error err = error("JWT token is expired");
-        return err;
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "JWT token is expired"});
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]:
```suggestion
        error jwtError = error(INTERNAL_ERROR_CODE, { message : "JWT token is expired" });
```

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal [hunk 11]
```diff
@@ -176,35 +193,48 @@ function parsePayload(json jwtPayloadJson) returns (JwtPayload) {
...
+        return jwtError;
     }
     if (!validateNotBeforeTime(jwtPayload)) {
-        error err = error("JWT token is used before Not_Before_Time");
-        return err;
+        error jwtError = error(INTERNAL_ERROR_CODE, { message : "JWT token is used before Not_Before_Time"});
```
Comment (Manuri) [inline stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]:
```suggestion
        error jwtError = error(INTERNAL_ERROR_CODE, { message : "JWT token is used before Not_Before_Time" });
```

File: stdlib/internal/src/main/ballerina/internal/jwt_validator.bal (hunk 3 above)
Comment (VijithaEkanayake) [reply stdlib/internal/src/main/ballerina/internal/jwt_validator.bal]: This error propagated from a native function in the internal package, So it has the same error code.

**Pull request merged**